fearless-utils/src/main/java/jp/co/soramitsu/fearless_utils/runtime/metadata/RuntimeBuilder.kt

```diff
@@ -0,0 +1,300 @@
+package jp.co.soramitsu.fearless_utils.runtime.metadata
+
+import jp.co.soramitsu.fearless_utils.runtime.definitions.registry.TypeRegistry
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.composite.DictEnum
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.composite.Struct
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.Constant
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.Error
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.Event
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.FunctionArgument
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.MetadataFunction
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.Module
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.Storage
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.StorageEntry
+import jp.co.soramitsu.fearless_utils.runtime.metadata.module.StorageEntryType
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.ExtrinsicMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.PalletCallMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.PalletConstantMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.PalletErrorMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.PalletEventMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.PalletMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.RuntimeMetadataSchemaV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.SignedExtensionMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.StorageEntryMetadataV14
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.StorageMetadataV14
+import jp.co.soramitsu.fearless_utils.scale.EncodableStruct
+import java.math.BigInteger
+
+object RuntimeBuilder {
```

Comment:
I suggest to have two implementations of the same interface for each runtime version (<14 and 14) and a factory which will return relevent one based on runtimeVersion. Such approach allow to get rid of mutiple ` when {
            v < 14 -> {` checks in this class